odl/trafos/util/ft_utils.py

```diff
@@ -0,0 +1,654 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utility functions for regular Fourier transforms."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import range
+
+import numpy as np
+
+from odl.discr import (RegularGrid, DiscreteLp, uniform_partition_fromgrid,
+                       uniform_discr_frompartition)
+from odl.set import RealNumbers
+from odl.util import (
+    fast_1d_tensor_mult,
+    is_real_dtype, is_scalar_dtype, is_real_floating_dtype,
+    is_complex_floating_dtype, dtype_repr, conj_exponent,
+    TYPE_MAP_R2C,
+    normalized_scalar_param_list)
+
+
+__all__ = ('reciprocal', 'inverse_reciprocal',
+           'reciprocal_space',
+           'dft_preprocess_data', 'dft_postprocess_data')
+
+
+def reciprocal(grid, shift=True, axes=None, halfcomplex=False):
```

Comment:
Could use being called `reciprocal_grid` now that it's not as obvious what it does.